python/cuml/common/base.pyx

```diff
@@ -292,6 +295,77 @@ class Base:
             return self.output_type
 
 
+class RegressorMixin:
+    """Mixin class for regression estimators in"""
+
+    _estimator_type = "regressor"
+
+    def score(self, X, y, **kwargs):
+        """Scoring function for linear classifiers
+
+        Returns the coefficient of determination R^2 of the prediction.
+
+        Parameters
+        ----------
+        X : array-like (device or host) shape = (n_samples, n_features)
+            Test samples on which we predict
+            Acceptable formats: cuDF DataFrame, NumPy ndarray, Numba device
+            ndarray, cuda array interface compliant array like CuPy
+        y : array-like (device or host) shape = (n_samples, n_features)
+            Ground truth values for predict(X)
+            Acceptable formats: cuDF DataFrame, NumPy ndarray, Numba device
+            ndarray, cuda array interface compliant array like CuPy
+
+        Returns
+        -------
+        score : float
+            R^2 of self.predict(X) wrt. y.
+        """
+        from cuml.metrics.regression import r2_score
+
+        if hasattr(self, 'handle'):
+            handle = self.handle
+        else:
+            handle = None
+
+        preds = self.predict(X)
+        return r2_score(y, preds, handle=handle)
```

Comment:
Should we increase compatibility by also overriding the `_more_tags()` method? https://github.com/scikit-learn/scikit-learn/blob/fd237278e/sklearn/base.py#L501. It

Reply:
I'm happy to implement this, though I wonder if it might be worth holding off unless there's a compelling reason. Do you know what this generally used for in sklearn @cjnolet ? It's not clear to me, but I'm not well versed in the sklearn internals.

Reply:
This was really a genuine question in my part. If you don’t see a need for this yet then I’m fine holding off. 

I just happened to notice this was the only difference between our mixins and those in Scikit-learn so I figured I’d ask. 

The scikit-learn documentation also gives little evidence into exactly how this is used. My first thinking was that it might be used for some type of tag-based model selection where only estimators meeting particular behavioral / characteristic criteria are trained and evaluated?

Reply:
I guess they are used primarily in tests and by helper functions to determine check/validate their inputs and outputs: https://scikit-learn.org/stable/developers/develop.html

Reply:
Nice find. Sharing the overview summary in this thread for readability:
```
Scikit-learn introduced estimator tags in version 0.21. These are annotations of estimators that allow programmatic inspection of their capabilities, such as sparse matrix support, supported output types and supported methods. The estimator tags are a dictionary returned by the method _get_tags().
``
Given your research into how sklearn uses them, I would vote we hold off for now. But I'm still happy to be persuaded otherwise

Reply:
I agree, I don't these are a high priority at the moment.